XMonad/Actions/CycleWS.hs

```diff
@@ -262,7 +279,43 @@ wsTypeToPred AnyWS      = return (const True)
 wsTypeToPred (WSTagGroup sep) = do cur <- groupName.workspace.current <$> gets windowset
                                    return $ (cur ==).groupName
                                    where groupName = takeWhile (/=sep).tag
-wsTypeToPred (WSIs p)   = p
+wsTypeToPred (WSIs p ) = p
+wsTypeToPred (p :&: q) = liftM2 (&&) <$> wsTypeToPred p <*> wsTypeToPred q
+wsTypeToPred (p :|: q) = liftM2 (||) <$> wsTypeToPred p <*> wsTypeToPred q
+wsTypeToPred (Not p  ) = fmap not <$> wsTypeToPred p
+
+
+emptyWS, hiddenWS, anyWS :: WSType
+
+-- | cycle through empty workspaces
```

Comment:
Given that these are all top-level haddocks now, I'd Capitalize the first word and put a period at the end for all of these

Reply:
True, I copied the documentation over from the constructors so that's way they're not capitalized